### ISSUE TITLE
Google Analytics Reporting API V4 cannot output metadata info & segment data as Core Reporting V3

### ISSUE THREAD

**Issue by DavidSUN2** [googleapis/google-api-python-client#1123]
Hi, I used the following code to output metadata info (e.g. accountID, propertyID) and segment data using Core Reporting V3:
```
from apiclient.discovery import build
from oauth2client.service_account import ServiceAccountCredentials

scope = 'https://www.googleapis.com/auth/analytics.readonly'
key_file_location = '********************.json'
viewid='********'

credentials = ServiceAccountCredentials.from_json_keyfile_name(key_file_location, scopes=scope)
service = build('analytics', 'v3', credentials=credentials)
results = service.data().ga().get(
            ids='ga:' + viewid,
            start_date='7daysAgo',
            end_date='today',
            metrics='ga:goal11Completions').execute()
print(results)   # output metadata such as accountID, propertyID

# the following chunk can output segment info
segments = service.management().segments().list().execute()  
for segment in segments.get('items', []):
    print('Segment segmentId  = %s' % segment.get('segmentId'))
    print('Segment type  = %s' % segment.get('type'))
    print('Segment Name       = %s' % segment.get('name'))
    print('Segment Definition = %s' % segment.get('definition'))

```
But when I change the API from v3 to v4 (i.e. service = build('analytics', 'v3', credentials=credentials) --> service= build('analyticsreporting', 'v4', credentials=credentials)), an error message pop up:
_AttributeError: 'Resource' object has no attribute 'data'
AttributeError: 'Resource' object has no attribute 'management'_

I felt frustrated because I have tried different ways of using V4 to output the similar results as V3, but still failed. So could you please give me some clue about why I failed using V4? Or currently V4 doesn't support this? Thanks

**Comment by parthea** [comment]
Hi @DavidSUN2 ,

I haven't worked with this API but I was able to take a look and see that in fact V3 and V4 are different. You will need to modify your code to be able to use the Analytics Reporting API v4 endpoint. [Here is a migration guide](https://developers.google.com/analytics/devguides/reporting/core/v4/migration#v4-endpoint) that shows the differences in the endpoints as well as [sample code](https://developers.google.com/analytics/devguides/reporting/core/v4/quickstart/service-py#3_setup_the_sample) for the Analytics Reporting API v4. 

For example,
instead of
```
service.data().ga().get()
```

you will need
```
analytics.reports().batchGet()
```

Please let me know if you have any questions!

**Comment by DavidSUN2** [comment]
Hi @parthea,

Thanks for your reply. And sorry for forgetting to mention that: I did change from service.data().ga().get() to service.reports().batchGet() and get the error message.

```
from apiclient.discovery import build
from oauth2client.service_account import ServiceAccountCredentials

scope = 'https://www.googleapis.com/auth/analytics.readonly'
key_file_location = '********************.json'
viewid='********'

credentials = ServiceAccountCredentials.from_json_keyfile_name(key_file_location, scopes=scope)
service = build('analyticsreporting', 'v4', credentials=credentials)
body={
'reportRequests': [
    { "viewId": viewid, 
    "pageSize": 300,
    "samplingLevel" : "LARGE",
    "dateRanges": [{"startDate": "2020-08-16", "endDate": "2020-08-17"}],
    "dimensions": [
    { "name": "ga:campaign"},
    { "name": "ga:dateHourMinute"}, 
    { "name": "ga:deviceCategory" }
    ], 
    "metrics": [
    { "expression": "ga:bounces" },
    { "expression": "ga:sessions" }
    ],
    "includeEmptyRows": "false" }]
}

results =  service.reports().batchGet(body=body).execute()
print(results) # output report data but not metadata, I cannot find accountID, propertyID info.

segments = service.management().segments().list().execute()  # error: service doesn't have management attribute
for segment in segments.get('items', []):
    print('Segment segmentId  = %s' % segment.get('segmentId'))
    print('Segment type  = %s' % segment.get('type'))
    print('Segment Name       = %s' % segment.get('name'))
    print('Segment Definition = %s' % segment.get('definition'))

```
So V4 cannot do what V3 does. I don't know why...

**Comment by parthea** [comment]
Hi @DavidSUN2,

TLDR; use the Analytics Management API v3 to list available segments.

Thanks for sharing additional information. I understand that you're looking for the `segments().list()` feature in the Analytics Reporting API v4. I did a bit of digging in the Analytics Reporting API v4 [samples](https://developers.google.com/analytics/devguides/reporting/core/v4/samples) and found the [segments section](https://developers.google.com/analytics/devguides/reporting/core/v4/samples#segments). At the very end of the section I see a note that says to use the Analytics Management API v3 to obtain a list of available segments.

> A list of available segments along with their description and segment id can be obtained using list method of Analytics Management API v3.

Since this note is in the Analytics Reporting API v4 samples page, it seems that the recommendation is to use Analytics Management API v3 to get list the segments. I also found a [StackOverflow post](https://stackoverflow.com/questions/38360241/how-to-get-a-list-of-views-profiles-in-google-analytics-api-v4) that mentions that Analytics Management API v3 and Analytics Reporting API v4 designed to be used together.

Here is a link to the public [Google Analytics Developers](https://issuetracker.google.com/issues/new?component=187400&template=765802) issue tracker in case you'd like to inquire further about the Google Analytics Reporting V4 API.

I'm going to close off this issue but please feel free to re-open it if you're still having trouble.
